### ISSUE TITLE
Slider improvements

### ISSUE THREAD

**Issue by sirinsidiator** [sirinsidiator/ESO-LibAddonMenu#49]
Some things that I have noticed while using sliders in AwesomeGuildStore:
- [x] option to select everything in the text input field when it gains focus
- [x] option to move the input field to the right side of the slider
- [x] option to allow input beyond min and max
- [x] slider does not update when input field looses focus without pressing enter
- [x] decimals option not working when calling UpdateValue

**Comment by sirinsidiator** [comment]
Added "autoselect" boolean. When it is true "OnFocusGained" is pre-hooked to select all text in the input box.

**Comment by sirinsidiator** [comment]
Added "inputLocation" option which accepts "right" or "below" (default). This option will put the input field on the right side and change the size. It should only be used for sliders in custom panels as it doesn't really look good in the default addon menu.

**Comment by sirinsidiator** [comment]
Added "clampInput" which defaults to true and if turned off allows values in the text input to go outside the range of the slider (specified by min and max). This also allows the mouse wheel to go change the value beyond these limits.

I also corrected the decimals option not working in some cases and the slider behavior when the text input field is used. It now will apply the value whenever the field looses focus, unless this happens because of the user pressing the escape key.
